File: leetcode/110-balanced-binary-tree/balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean flag = true;
    public boolean isBalanced(TreeNode root) {
        if(root == null) return true;
         int t = fun(root);
         return flag;
    }
    public int fun(TreeNode root)
    {
        if(root == null) return 0;

        int l = fun(root.left);
        int r = fun(root.right);

        if(Math.abs(l - r) > 1)
        flag = false;

        return Math.max(l , r) + 1;


    }
}

File: leetcode/110-balanced-binary-tree/balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
 var flag ;
var isBalanced = function(root) {
    flag = true;
    check(root);
    return flag;
};
var check = function(root){
    if(root === null) return 0;
    let left = check(root.left);
    let right = check(root.right);
    if(Math.abs(left - right) > 1){
        flag = false;
    }

    return Math.max(left,right) + 1;
}